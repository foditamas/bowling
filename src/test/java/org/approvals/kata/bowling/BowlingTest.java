package org.approvals.kata.bowling;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.UseReporter;

@UseReporter(org.approvaltests.reporters.JunitReporter.class)
public class BowlingTest extends TestCase
{
  /**
    * For this bowling kata, try to get this test to pass using these simple rules:
    * 
    * 1)  run this test often
    * 2)  don't write any other tests
    * 
    * note: you will need to create a toString() method on the Bowling class
    * 
    * How To Score Bowling : http://en.wikipedia.org/wiki/Ten-pin_bowling#Scoring_2
    * 
    * Things to notice
    * *   how good was your feedback ?
    * *   what was your test to code ratio?
    * *   were failing tests informative?
    * *   did you know when you were done?
    * *   did you know what to do next?
    * *   what was test coverage?
    * 
    */
  public void testBowlingGame() throws Exception
  {
    Bowling bowling = new Bowling();
    bowling.rolls(5, 3, 6, 4, 2, 7, 10, 4, 5, 3, 7, 10, 10, 8, 1, 10, 10, 10);
    Approvals.verify(bowling.toString());
  }
}
