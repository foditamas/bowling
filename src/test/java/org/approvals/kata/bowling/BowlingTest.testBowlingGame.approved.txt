1) 5, 3 [8] => 8 
2) 6, 4 [12] => 20 
3) 2, 7 [9] => 29 
4) 10 [19] => 48 
5) 4, 5 [9] => 57 
6) 3, 7 [20] => 77 
7) 10 [28] => 105 
8) 10 [19] => 124 
9) 8, 1 [9] => 133 
10) 10, 10, 10 [30] => 163 
